IHP/IDE/SchemaDesigner/Parser.hs

```diff
@@ -172,6 +172,34 @@ parseCheckConstraint name = do
     checkExpression <- between (char '(' >> space) (char ')' >> space) expression
     pure CheckConstraint { name, checkExpression }
 
+parseExcludeConstraint name = do
+    lexeme "EXCLUDE"
+    excludeElements <- between (char '(' >> space) (char ')' >> space) $ excludeElement `sepBy` (char ',' >> space)
+    predicate <- optional do
+        lexeme "WHERE"
+        between (char '(' >> space) (char ')' >> space) expression
+    pure ExcludeConstraint { name, excludeElements, predicate }
+    where
+        excludeElement = do
+            element <- identifier
+            space
+            lexeme "WITH"
+            space
+            operator <- parseOperator
+            pure ExcludeConstraintElement { element, operator }
+
+parseOperator = choice $ map lexeme
```

Comment:
Maybe we can call this `parseExcludeOperator` to avoid this being confused with the general operator parsing logic (which handles infix, prefix, postfix etc.)?

```suggestion
parseExcludeOperator = choice $ map lexeme
```